client/src/reducers/feedReducers.js

```diff
@@ -0,0 +1,41 @@
+import {
+  ADD_OPTION,
+  REMOVE_OPTION,
+  REMOVE_ALL_OPTIONS,
+  TOGGLE_STATE,
+  SET_VALUE,
+} from "../actions/feedActions";
+
+export const feedReducer = (oldState, action) => {
+  let newState = Object.assign({}, oldState);
+  const { type, key, value } = action;
+  switch (type) {
+    case TOGGLE_STATE:
+      newState[key] = !newState[key];
+      return newState;
+    case SET_VALUE:
+      newState[key] = value;
+      return newState;
+    default:
+      return oldState;
+  }
+};
+
+export const optionsReducer = (oldState, action) => {
+  const { option, label } = action.payload;
+  let newState = Object.assign({}, oldState);
+  newState[label] = newState[label] || [];
+  switch (action.type) {
+    case ADD_OPTION:
+      newState[label].push(option);
```

Comment:
if you clone the `oldState` here and it keeps reference to the `oldState[label]` array your should also immutably update the nested object and arrays, so not using `[].push()` but e.g. 
```
return { ...oldState, [label]: [ ...oldState[label], option ] };
```
more examples here https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns